### PR TITLE
e2e: Test against multiple versions of k8s at the same time

### LINKED ISSUE #1454: Bump kubernetes dependencies to v1.21 version
Couple of things to do:

* In master branch, we want to bump client-go & co to `v0.21` -> https://github.com/kubernetes/kube-state-metrics/blob/master/go.mod#L41-L43

* We also need to test against Kubernetes v1.21, for that we should change this to v1.21 https://github.com/kubernetes/kube-state-metrics/blob/7fd83751066b39d6f80fd31ee190d1cd2cd6352a/tests/e2e.sh#L27 maybe even adding the ability to test against both v1.20 and v1.21. cc @mrueg what do you think?

* Document compatibility in the README file.

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
We support multiple versions of k8s, let's run e2e tests for all as suggested by @lilic 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
See #1454

### CODE REVIEW

**Review by lilic — APPROVED** [review]
Nice thank you! 🎉 
/lgtm

/hold in case any objections from other folks

**Comment by dgrisonnet** [comment]
Just a side note, if this PR is merged as is, it will close #1454 because `resolves` was specified in the description even though it was meant to be a partial resolution.

File: tests/e2e.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ case $(uname -m) in
 esac
 
 NODE_IMAGE_NAME="docker.io/kindest/node"
-KUBERNETES_VERSION=v1.20.0
+KUBERNETES_VERSION=${KUBERNETES_VERSION:-v1.20.0}
```
Comment (dgrisonnet) [inline tests/e2e.sh:27]: We might want to change the default to `v1.21.0` now that it has been released.

**Comment by dgrisonnet** [comment]
It might also be meaningful to update the compatibility matrix as part of this effort since it verifies that kube-state-metrics is compatible with Kubernetes `v1.21.0`

File: tests/e2e.sh (hunk 1 above)
Comment (mrueg) [reply tests/e2e.sh]: Done thanks!

**Comment by tariq1890** [comment]
/hold cancel

Thanks @mrueg !

**Pull request merged**